module/zcommon/zfs_prop.c

```diff
@@ -1016,7 +1016,12 @@ zcommon_fini(void)
 	kfpu_fini();
 }
 
+#ifdef __FreeBSD__
+SYSINIT(zfs_zcommon_init, SI_SUB_INT_CONFIG_HOOKS, SI_ORDER_FIRST,
+    zcommon_init, NULL);
+#else
```

Comment:
Let's do this with a new wrapper instead.  How about something like `module_init_first()`?  Or would it make sense to just make this the default for `module_init()`?

Reply:
I think part of what makes this work is that it now runs before the other things that are run with `module_init()`, so a `module_init_early()` wrapper makes more sense here.